projects/rocprofiler-systems/source/lib/core/trace_cache/perfetto_processor.cpp

```diff
@@ -0,0 +1,1142 @@
+    {
+        auto make_track_name = [&](const char* metric) {
+            return JOIN(" ", "GPU", JOIN("", '[', _device_id, ']'), metric, "(S)");
+        };
+
+        if(!amd_smi_pcie_link_width_track::exists(_device_id))
+        {
+            amd_smi_pcie_link_width_track::emplace(
+                _device_id, make_track_name("PCIe Link Width"), "");
+        }
+        TRACE_COUNTER("device_pcie_link_width",
+                      amd_smi_pcie_link_width_track::at(_device_id, 0), _ts,
+                      static_cast<double>(gpu_metrics.pcie_link_width));
+
+        if(!amd_smi_pcie_link_speed_track::exists(_device_id))
+        {
+            amd_smi_pcie_link_speed_track::emplace(
+                _device_id, make_track_name("PCIe Link Speed"), "MT/s");
+        }
+        TRACE_COUNTER("device_pcie_link_speed",
+                      amd_smi_pcie_link_speed_track::at(_device_id, 0), _ts,
+                      static_cast<double>(gpu_metrics.pcie_link_speed));
+
+        if(!amd_smi_pcie_bandwidth_acc_track::exists(_device_id))
+        {
+            amd_smi_pcie_bandwidth_acc_track::emplace(
+                _device_id, make_track_name("PCIe Bandwidth Acc"), "bytes");
+        }
+        TRACE_COUNTER("device_pcie_bandwidth_acc",
+                      amd_smi_pcie_bandwidth_acc_track::at(_device_id, 0), _ts,
+                      static_cast<double>(gpu_metrics.pcie_bandwidth_acc));
+
+        if(!amd_smi_pcie_bandwidth_inst_track::exists(_device_id))
+        {
+            amd_smi_pcie_bandwidth_inst_track::emplace(
+                _device_id, make_track_name("PCIe Bandwidth Inst"), "bytes");
+        }
+        TRACE_COUNTER("device_pcie_bandwidth_inst",
+                      amd_smi_pcie_bandwidth_inst_track::at(_device_id, 0), _ts,
+                      static_cast<double>(gpu_metrics.pcie_bandwidth_inst));
+    }
+}
+
+void
+perfetto_processor_t::handle([[maybe_unused]] const in_time_sample& _sample)
+{
+    auto event_metadata = nlohmann::json::parse(_sample.event_metadata);
+
+    auto _track_name = _sample.track_name;
+    auto _timestamp  = _sample.timestamp_ns;
+
+    std::string _name       = event_metadata.value("name", "");
+    std::string _event_type = event_metadata.value("event_type", "");
+    std::string _target     = event_metadata.value("target", "");
+
+    auto _track_uuid = std::hash<std::string>{}(_track_name);
+
+    auto _track = get_track(category::kokkos{}, _track_name, _track_uuid);
+    auto add_perfetto_annotations = [&](::perfetto::EventContext ctx) {
+        if(!m_use_annotations) return;
+
+        annotate_perfetto(ctx, { { "timestamp_ns", _timestamp },
+                                 { "event_type", _event_type },
+                                 { "target", _target } });
+    };
+
+    TRACE_EVENT_INSTANT(trait::name<category::kokkos>::value,
+                        ::perfetto::DynamicString{ _name }, _track, _timestamp,
+                        add_perfetto_annotations);
```

Comment:
```suggestion
    const auto event_metadata = nlohmann::json::parse(_sample.event_metadata);

    const auto _track_name = _sample.track_name;
    const auto _timestamp  = _sample.timestamp_ns;

    const std::string _event_name       = event_metadata.value("name", "");
    const std::string _event_type = event_metadata.value("event_type", "");
    const std::string _event_target     = event_metadata.value("target", "");

    const auto _track_uuid = std::hash<std::string>{}(_track_name);

    auto _track = get_track(category::kokkos{}, _track_name, _track_uuid);
    auto add_perfetto_annotations = [&](::perfetto::EventContext ctx) {
        if(!m_use_annotations) return;

        annotate_perfetto(ctx, { { "timestamp_ns", _timestamp },
                                 { "event_type", _event_type },
                                 { "target", _event_target } });
    };

    TRACE_EVENT_INSTANT(trait::name<category::kokkos>::value,
                        ::perfetto::DynamicString{ _event_name }, _track, _timestamp,
                        add_perfetto_annotations);
```
Do we is really need `category::kokkos` hardcoded?

Reply:
`kokkos` is the only data we handle as an `in_time_sample`, and that was the sole reason why it was hardcoded. I don't think we need it hardcoded here, I am not sure why I left it as such. Will look into it and apply fix, if possible.